### PR TITLE
✨ Add @umpire/tanstack-form — TanStack Form adapter with React, Solid, and Vue renderers

### PR DESCRIPTION
**Pull request by sdougbrown**
## `@umpire/tanstack-form` — TanStack Form Adapter

New package bridging Umpire's declarative field-availability graph into TanStack Form validation and conditional rendering. Ships with React, Solid, and Vue renderers.

### What this does

TanStack Form has no native model for conditional field relationships. Blog posts about this devolve into imperative `form.subscribe` + manual dependency tracking. This package lets you declare the field graph once in Umpire and have it drive validation, linked-field plumbing, conditional rendering, and strike suggestions — across all three renderers.

### Package shape

```
packages/tanstack-form/
  src/
    adapter.ts       — createUmpireFormAdapter (framework-neutral)
    dependencies.ts  — getUmpireLinkedFields (graph → listenTo plumbing)
    options.ts       — createUmpireFormOptions (strike, listeners, conditions)
    reads.ts         — umpireReadListeners (side-effect wiring)
    dynamic.ts       — umpireDynamicValidator (form-level escape hatch)
    validator.ts     — umpireFieldValidator / umpireFieldValidators
    react.tsx        — useUmpireForm, UmpireFormSubscribe, createUmpireFormComponents
    solid.tsx        — createUmpireForm, UmpireFormSubscribe, createUmpireFormComponents
    vue.ts           — useUmpireForm, UmpireFormSubscribe
    index.ts         — framework-neutral re-exports
```

### Exports

| Entry | What ships |
|-------|-----------|
| `@umpire/tanstack-form` | Framework-neutral validators, adapter, options, linked-field plumbing, reads |
| `@umpire/tanstack-form/react` | `useUmpireForm`, `UmpireFormSubscribe`, `createUmpireFormComponents` |
| `@umpire/tanstack-form/solid` | `createUmpireForm`, `UmpireFormSubscribe`, `createUmpireFormComponents` |
| `@umpire/tanstack-form/vue` | `useUmpireForm`, `UmpireFormSubscribe` |

### Key design decisions

- **Field validation is the primary path.** `umpireFieldValidator` plugs into field-level `validators.onChange` — errors land in `field.state.meta.errors` where every standard field component reads from them. `umpireDynamicValidator` is an escape hatch that places errors in `form.state.errorMap.onDynamic` (documented prominently).
- **Linked fields are graph-derived, not hand-written.** `getUmpireLinkedFields` reads `engine.graph().edges` and generates `onChangeListenTo` / `onBlurListenTo` automatically. Explicit `listenTo` override is available.
- **`createFormHook` composition is first-class.** React and Solid ship `createUmpireFormComponents` returning `UmpireScope` / `UmpireField` / `UmpireSubmit` for registration with TanStack's `createFormHook` pattern. Vue skips this since `createFormHook` doesn't exist in the Vue adapter.
- **Strike applies `foul.suggestedValue`, not `resetField`.** Matches Umpire's `strike()` semantics. `resetField` is available as an opt-in mode.
- **Type source is `@tanstack/form-core`.** Renderer-neutral — no React/Solid/Vue in its exports. No local structural types needed.

### Coverage

98 tests across 9 files covering all framework-neutral modules, all three renderers, edge cases for self-reference exclusion, deduplication, `rejectFoul` toggling, event selection, snapshot isolation, and conditions as functions/plain objects.

### Docs

New Starlight guide at `/adapters/tanstack-form` with per-framework usage sections.

### Peer deps

- Root: `@tanstack/form-core`
- React: `react`, `@tanstack/react-form`
- Solid: `solid-js`, `@tanstack/solid-form`
- Vue: `vue`, `@tanstack/vue-form`

### CODE REVIEW

File: packages/tanstack-form/src/validator.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        }
+      >
+    )[fieldName]
+
+    if (!fieldCheck?.enabled) return undefined
+    if (options?.rejectFoul !== false && fieldCheck.fair === false) {
+      return fieldCheck.reason ?? 'Invalid value'
+    }
+    if (fieldCheck.error) return fieldCheck.error
+
```
[the remark this answers is not in the document]
Comment (sdougbrown) [reply packages/tanstack-form/src/validator.ts:57]: `👾 AI Agent`

Addressed. `umpireFieldValidator` now mirrors the dynamic validator for required fields: enabled + required + unsatisfied returns the field reason or `Required`, and there is regression coverage for the null required-field case.

File: packages/tanstack-form/src/validator.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+          error?: string
+        }
+      >
+    )[fieldName]
+
+    if (!fieldCheck?.enabled) return undefined
+    if (options?.rejectFoul !== false && fieldCheck.fair === false) {
+      return fieldCheck.reason ?? 'Invalid value'
+    }
+    if (fieldCheck.error) return fieldCheck.error
```
[the remark this answers is not in the document]
Comment (sdougbrown) [reply packages/tanstack-form/src/validator.ts:56]: `👾 AI Agent`

Applied. Field-level validation now checks `required && !satisfied` before foul/error handling, matching the form-level validator behavior.

File: packages/tanstack-form/src/adapter.ts [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+  function getFouls(): Foul<F>[] {
+    const current: Snapshot<C> = {
+      values: snapshotValue(form.state.values),
+      conditions: snapshotValue(getConditions()) as C | undefined,
+    }
+
+    if (!previousSnapshot) {
+      previousSnapshot = current
+      return []
```
[the remark this answers is not in the document]
Comment (sdougbrown) [reply packages/tanstack-form/src/adapter.ts]: `👾 AI Agent`

Addressed. `getFouls()` now caches the latest computed transition, and `applyStrike()` reuses that cached transition when the current snapshot matches instead of advancing the snapshot again. `applyStrike()` also clears the cache after applying so repeated calls do not replay the same strikes.

File: packages/tanstack-form/src/adapter.ts [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+  C extends Record<string, unknown>,
+>(
+  form: {
+    state: { values: Record<string, unknown> }
+    setFieldValue(name: string, value: unknown): void
+  },
+  engine: Umpire<F, C>,
+  options?: UmpireFormAdapterOptions<C>,
+): UmpireFormAdapter<F> {
+  let previousSnapshot: Snapshot<C> | null = null
```
[the remark this answers is not in the document]
Comment (sdougbrown) [reply packages/tanstack-form/src/adapter.ts:54]: `👾 AI Agent`

Applied with the rest of the adapter fix. The adapter now tracks the cached snapshot and cached fouls so `applyStrike()` can consume an inspected foul list safely.

File: packages/tanstack-form/src/adapter.ts [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+    previousSnapshot = current
+    return fouls
+  }
+
+  function applyStrike(): void {
+    const fouls = getFouls()
+    for (const foul of fouls) {
+      setFieldValue(foul.field, foul.suggestedValue)
+    }
+  }
```
[the remark this answers is not in the document]
Comment (sdougbrown) [reply packages/tanstack-form/src/adapter.ts:128]: `👾 AI Agent`

Addressed. I used a slightly stricter version of this fix: `applyStrike()` reuses cached fouls only when they correspond to the current snapshot, computes fresh fouls when values changed, and clears the cache after applying to avoid duplicate strikes.
